### ISSUE TITLE
Anito: Defend a Land Enraged

### ISSUE THREAD

**Issue by huckleberrypie** [AlpyneDreams/d8vk#71]
The game crashes on modern Windows versions unless you use dgVoodoo in WARP/software rendering mode. I was hoping d8vk would come to the rescue but alas.

Since this game isn't that well-known, a free demo of the game is available here: https://download.cnet.com/Anito-Defend-a-Land-Enraged-demo/3000-7536_4-10255734.html

### Software information
Anito: Defend a Land Enraged

### System information
- GPU: AMD Radeon RX 570
- Driver: Adrenalin 22.11.1
- Wine version: N/A
- DXVK version: 0.1.0

### Apitrace file(s)
- [Anito.zip](https://github.com/AlpyneDreams/d8vk/files/10163757/Anito.zip)

For instructions on how to use apitrace, see: https://github.com/doitsujin/dxvk/wiki/Using-Apitrace

### Log files
- d3d9.log: [Anito_d3d9.log](https://github.com/AlpyneDreams/d8vk/files/10163751/Anito_d3d9.log)

**Comment by Blisto91** [comment]
I haven't checked out your trace or the demo yet, but remember to use the CI artifacts of latest master if you aren't already as the release build is heavily outdated.
You can find the latest builds here 🙂 
https://github.com/AlpyneDreams/d8vk/actions/workflows/artifacts.yml?query=branch%3Amaster

**Comment by huckleberrypie** [comment]
> I haven't checked out your trace or the demo yet, but remember to use the CI artifacts of latest master if you aren't already as the release build is heavily outdated. You can find the latest builds here 🙂 https://github.com/AlpyneDreams/d8vk/actions/workflows/artifacts.yml?query=branch%3Amaster

Still no dice tho, the game still crashes to the desktop with the following log: [Anito_d3d9.log](https://github.com/AlpyneDreams/d8vk/files/10164052/Anito_d3d9.log)

**Comment by AlpyneDreams** [comment]
> Still no dice tho, the game still crashes to the desktop with the following log: [Anito_d3d9.log](https://github.com/AlpyneDreams/d8vk/files/10164052/Anito_d3d9.log)

Can you try capturing a d3d8 apitrace with this version? Thanks for reporting this bug :)

**Comment by huckleberrypie** [comment]
> > Still no dice tho, the game still crashes to the desktop with the following log: [Anito_d3d9.log](https://github.com/AlpyneDreams/d8vk/files/10164052/Anito_d3d9.log)
> 
> Can you try capturing a d3d8 apitrace with this version? Thanks for reporting this bug :)

Exactly which files should I copy from the apitrace archive to get a trace?

**Comment by huckleberrypie** [comment]
Hello?

**Comment by Blisto91** [comment]
Should just be the d3d8 dll

**Comment by huckleberrypie** [comment]
> Should just be the d3d8 dll

Alright, let me try again.

**Comment by huckleberrypie** [comment]
Here's the trace you requested: 
[Anito.zip](https://github.com/AlpyneDreams/d8vk/files/10313877/Anito.zip)

**Comment by huckleberrypie** [comment]
Has anyone looked into said trace?

**Comment by Blisto91** [comment]
Not yet. Alpyne has been busy with school.
Development should resume soon afaik

**Comment by huckleberrypie** [comment]
> Not yet. Alpyne has been busy with school. Development should resume soon afaik

I presume he'll look into it now though?

**Comment by Blisto91** [comment]
Yes when he's got the time i assume.

**Comment by huckleberrypie** [comment]
> Yes when he's got the time i assume.

Hope he'll tend to this soon, but it's no biggie as I am not in a rush.

**Comment by huckleberrypie** [comment]
Re-opening this as I am somehow getting this error log upon trying the latest artefact:

info:  Game: Anito.exe
info:  DXVK: d8vk-v0.10-9-g6162ce42
info:  Found built-in config:
info:    d3d9.maxAvailableMemory = 1024
info:    d3d9.memoryTrackTest = True
info:  Effective configuration:
info:    d3d9.memoryTrackTest = True
info:    d3d9.maxAvailableMemory = 1024
info:  Built-in extension providers:
info:    Win32 WSI
info:    OpenVR
info:    OpenXR
info:  OpenVR: could not open registry key, status 2
info:  OpenVR: Failed to locate module
info:  Enabled instance extensions:
info:    VK_KHR_get_surface_capabilities2
info:    VK_KHR_surface
info:    VK_KHR_win32_surface
info:  D3D9: VK_FORMAT_D24_UNORM_S8_UINT -> VK_FORMAT_D32_SFLOAT_S8_UINT
info:  Radeon RX 570 Series:
info:    Driver: 2.0.233
info:    Vulkan: 1.3.217
info:    Memory Heap[0]: 
info:      Size: 3840 MiB
info:      Flags: 0x3
info:      Memory Type[0]: Property Flags = 0x1
info:    Memory Heap[1]: 
info:      Size: 7892 MiB
info:      Flags: 0x0
info:      Memory Type[1]: Property Flags = 0x6
info:      Memory Type[3]: Property Flags = 0xe
info:    Memory Heap[2]: 
info:      Size: 256 MiB
info:      Flags: 0x3
info:      Memory Type[2]: Property Flags = 0x7
info:  Process set as DPI aware

Both demo and retail versions show the exact same error and crash silently.

**Comment by WinterSnowfall** [comment]
Sorry to hear that. I fixed it for Linux and can confirm it now works in Wine, but I guess Windows still has problems of its own. @AlpyneDreams Guess we can reopen this and add the Windows label.

**Comment by huckleberrypie** [comment]
> Sorry to hear that. I fixed it for Linux and can confirm it now works in Wine, but I guess Windows still has problems of its own. @AlpyneDreams Guess we can reopen this and add the Windows label.

Could it be a driver issue of some kind?

**Comment by WinterSnowfall** [comment]
> Could it be a driver issue of some kind?

It's most likely a d8vk issue that's specific to Windows. There are other games that have been confirmed to work on Linux and silently crash on Windows, such as Advent Rising.

The somewhat good news is that everything works fine in Wine, including the features that the game lists as hardware accelerated (shadows and transparency IIRC).

**Comment by huckleberrypie** [comment]
> > Could it be a driver issue of some kind?
> 
> It's most likely a d8vk issue that's specific to Windows. There are other games that have been confirmed to work on Linux and silently crash on Windows, such as Advent Rising.
> 
> The somewhat good news is that everything works fine in Wine, including the features that the game lists as hardware accelerated (shadows and transparency IIRC).

I see. I'll look forward to the fix soon.

**Comment by WinterSnowfall** [comment]
@huckleberrypie Could you please retest the game with the latest artifacts? d8vk has switched to linking d3d9 statically recently, so this means you'll only have d3d8.dll to worry about now. This may have improved the situation on Windows, hopefully.

**Comment by huckleberrypie** [comment]
> @huckleberrypie Could you please retest the game with the latest artifacts? d8vk has switched to linking d3d9 statically recently, so this means you'll only have d3d8.dll to worry about now. This may have improved the situation on Windows, hopefully.

How do I download the artefacts again?

EDIT: NVM, downloaded the latest artefacts and I can confirm that the game now works as it should, thanks!

**Comment by WinterSnowfall** [comment]
Great, happy to hear that it's working now :tada:. @AlpyneDreams The issue can be (re-)closed.
